sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -114,7 +114,29 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
 
   lazy val containsChild: Set[TreeNode[_]] = children.toSet
 
-  private lazy val _hashCode: Int = scala.util.hashing.MurmurHash3.productHash(this)
```

Comment:
so Scala doesn't fix `scala.util.hashing.MurmurHash3.productHash`?

Reply:
Scala 2.12 didn't fix: 
https://github.com/scala/scala/blob/2.12.x/src/library/scala/util/hashing/MurmurHash3.scala#L61-L79

but Scala 2.13 fixed it:
https://github.com/scala/scala/blob/v2.13.1/src/library/scala/util/hashing/MurmurHash3.scala#L56-L73

Reply:
`scala.util.hashing.MurmurHash3.productHash` calls the same implementations linked above.

Reply:
can we add a comment? When we drop scala 2.12 and move to scala 2.13, we can remove this.

Reply:
Yup.